crates/ruff/src/rules/ruff/typing.rs

```diff
@@ -135,7 +101,12 @@ impl<'a> TypingTarget<'a> {
                     )
                 }
             }
-            Expr::BinOp(..) => Some(TypingTarget::PEP604Union(expr)),
+            Expr::BinOp(ast::ExprBinOp {
+                left,
+                op: Operator::BitOr,
+                right,
+                ..
+            }) => Some(TypingTarget::PEP604Union(left, right)),
```

Comment:
I wonder if we should just short-circuit if the `op` isn't `BitOr` and return `Unknown`? 

Just thinking out loud that otherwise this will try to `resolve_call_path` and  return `Unknown` type. Either ways, `contains_none` will be `true` and `contains_any` will be `false`.

Reply:
I think it's fine because `resolve_call_path` will exit "immediately" when it detects the expression type.